v2/bigquery-to-bigtable/README.md

```diff
@@ -0,0 +1,162 @@
+        "helpText": "BigQuery query to export data from",
+        "paramType": "TEXT",
+        "isOptional": false
+      },
+      {
+        "name": "readIdColumn",
+        "label": "BigQuery query unique column ID",
+        "helpText": "BigQuery query unique column ID",
+        "paramType": "TEXT",
+        "isOptional": false
+      },
+      {
+        "name": "bigtableWriteProjectId",
+        "label": "Bigtable project id",
+        "helpText": "Bigtable project id to write to",
+        "paramType": "TEXT",
+        "isOptional": false
+      },
+      {
+        "name": "bigtableWriteInstanceId",
+        "label": "Bigtable instance id",
+        "helpText": "Bigtable instance id to write to",
+        "paramType": "TEXT",
+        "isOptional": false
+      },
+      {
+        "name": "bigtableWriteAppProfile",
+        "label": "Bigtable app profile",
+        "helpText": "Bigtable app profile to use for the export",
+        "paramType": "TEXT",
+        "isOptional": true
+      },
+      {
+        "name": "bigtableWriteTableId",
+        "label": "Bigtable table id",
+        "helpText": "Bigtable table id to write to",
+        "paramType": "TEXT",
+        "isOptional": false
+      },
+      {
+        "name": "bigtableWriteColumnFamily",
+        "label": "Bigtable table column family",
+        "helpText": "Bigtable table column family id to write to",
+        "paramType": "TEXT",
+        "isOptional": false
+      }
+    ]
+  },
+  "sdk_info": {
+    "language": "JAVA"
+  }
+}' > image_spec.json
+gsutil cp image_spec.json ${TEMPLATE_IMAGE_SPEC}
+rm image_spec.json
+```
+
+### Testing Template
+
+The template unit tests can be run using:
+```sh
+mvn test
+```
+
+### Executing Template
+
+The template requires the following parameters:
+* readQuery: BigQuery query to export data from
+* readIdColumn: BigQuery query unique column ID
+* bigtableWriteProjectId: Bigtable project id to write to
+* bigtableWriteInstanceId: Bigtable instance id to write to
+* bigtableWriteTableId: Bigtable table id to write to
+* bigtableWriteColumnFamily: Bigtable table column family id to write to
+
+The template has the following optional parameters:
+* bigtableWriteAppProfile: Bigtable app profile to use for the export
+
+Template can be executed using the following gcloud command:
+```sh
+export JOB_NAME="${TEMPLATE_MODULE}-`date +%Y%m%d-%H%M%S-%N`"
+gcloud beta dataflow flex-template run ${JOB_NAME} \
+        --project=${PROJECT} --region=${REGION} \
+        --template-file-gcs-location=${TEMPLATE_IMAGE_SPEC} \
+        --parameters ^~^readQuery="${READ_QUERY}" \
+        --parameters readIdColumn=${READ_ID_COLUMN},bigtableWriteProjectId=${BIGTABLE_WRITE_PROJECT_ID},bigtableWriteInstanceId=${BIGTABLE_WRITE_INSTANCE_ID},bigtableWriteTableId=${BIGTABLE_WRITE_TABLE_ID},bigtableWriteColumnFamily=${BIGTABLE_WRITE_COLUMN_FAMILY}
+```
+
+Note: The `^~^` prefix on readQuery indicates `~` as a delimiter for the query. If your
```

Comment:
I think this should be reworded to make it clear that '~' is used to make sure that commas aren't treated as delimiters. Something like: "The `^~^` prefix on `readQuery` is used to make `~` a delimiter instead of commas. This allows commas to be used in the query."

I think that makes it clearer why it is needed. It'll also help explain why something other than `~` should be used if it's in the query or indicate cases where the custom delimiter is unnecessary (e.g. `SELECT * FROM foo.bar`).

Maybe also add a link to the custom delimiter page so that people can get more info on this if they aren't familiar with this feature of `gcloud`.

Reply:
great, updated